src/mongo/util/net/httpclient.cpp

```diff
@@ -85,8 +86,8 @@ int HttpClient::_go(const char* command, string url, const char* body, Result* r
     string::size_type idx = host.find(":");
     if (idx != string::npos) {
         server = host.substr(0, idx);
-        string t = host.substr(idx + 1);
-        port = atoi(t.c_str());
+        const string t = host.substr(idx + 1);
+        parseNumberFromString<int>(t.c_str(), &port);
```

Comment:
Here's another place where the value of port will be different when the string does not parse in your change vs in the original code. Rather than match the old behavior, here, you could simply convert the bad status to a DBException as follows:

uassertStatusOK(parseNumberFromString<int>(t, &port));

This is OK here, because this method already clearly uses uassert to throw exceptions on errors.

Note also that you don't need or want to call c_str on strings passed into parseNumberFromString, as they can operate on std::string.